packages/simcore-sdk/src/simcore_sdk/node_ports_common/aws_s3_cli.py

```diff
@@ -0,0 +1,307 @@
+):
+    while True:
+        line: bytes = await stream.readline()
+        if line:
+            decoded_line = line.decode()
+            await logged_gather(
+                *[parser(decoded_line) for parser in aws_s3_cli_log_parsers]
+            )
+        else:
+            break
+
+
+@cached()
+async def is_aws_s3_cli_available(aws_s3_cli_settings: AwsS3CliSettings | None) -> bool:
+    """returns: True if the `aws` cli is installed and a configuration is provided"""
+    if aws_s3_cli_settings is None:
+        return False
+    try:
+        await _async_aws_cli_command(
+            "aws", "--version", aws_s3_cli_settings=aws_s3_cli_settings
+        )
+        return True
+    except AwsS3CliFailedError:
+        return False
+
+
+async def _async_aws_cli_command(
+    *cmd: str,
+    aws_s3_cli_settings: AwsS3CliSettings,
+    aws_cli_s3_log_parsers: list[BaseLogParser] | None = None,
+) -> str:
+    str_cmd = " ".join(cmd)
+    proc = await asyncio.create_subprocess_shell(
+        str_cmd,
+        stdin=asyncio.subprocess.PIPE,
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.STDOUT,
+        env={
+            "AWS_ACCESS_KEY_ID": aws_s3_cli_settings.AWS_S3_CLI_S3.S3_ACCESS_KEY,
+            "AWS_SECRET_ACCESS_KEY": aws_s3_cli_settings.AWS_S3_CLI_S3.S3_SECRET_KEY,
+        },
+    )
+
+    command_result_parser = CommandResultCaptureParser()
+    aws_cli_s3_log_parsers = (
+        [*aws_cli_s3_log_parsers, command_result_parser]
+        if aws_cli_s3_log_parsers
+        else [command_result_parser]
+    )
+
+    assert proc.stdout  # nosec
+    await asyncio.wait([_read_stream(proc.stdout, [*aws_cli_s3_log_parsers])])
+
+    _stdout, _stderr = await proc.communicate()
+
+    command_output = command_result_parser.get_output()
+    if proc.returncode != 0:
+        raise AwsS3CliFailedError(
+            command=str_cmd,
+            command_output=command_output,
+            returncode=proc.returncode,
+        )
+
+    _logger.debug("'%s' result:\n%s", str_cmd, command_output)
+    return command_output
+
+
+def _get_exclude_filters(exclude_patterns: set[str] | None) -> list[str]:
+    if exclude_patterns is None:
+        return []
+
+    exclude_options: list[str] = []
+    for entry in exclude_patterns:
+        exclude_options.append("--exclude")
+        exclude_options.append(entry.replace("*", "**"))
+
+    return exclude_options
+
+
+async def _get_s3_folder_size(
+    aws_s3_cli_settings: AwsS3CliSettings,
+    *,
+    s3_path: str,
+) -> ByteSize:
+    cli_command = [
+        "aws",
+        "s3",
+        "ls",
+        "--summarize",
+        "--recursive",
+        s3_path,
+        "| grep 'Total Size' | awk '{print $3}'",
+    ]
+
+    if aws_s3_cli_settings.AWS_S3_CLI_PROVIDER.value != S3Provider.AWS:
+        cli_command.insert(
+            1, f"--endpoint-url {aws_s3_cli_settings.AWS_S3_CLI_S3.S3_ENDPOINT}"
+        )
```

Comment:
Here I would recommend to re-use the same pattern we do use for all the other AWS based tools (see in aws-library and the associated settings)
The endpoint is None if not using AWS, therefore there should be no need for that S3Provider. see https://github.com/ITISFoundation/osparc-simcore/blob/5884e48f7c845f5050942fe0be32dad2bc592139/packages/settings-library/src/settings_library/s3.py